### PR TITLE
#114 fix string equality

### PR DESCRIPTION
**Pull request by y9san9**

### CODE REVIEW

File: resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt [hunk 1]
```diff
@@ -22,4 +21,13 @@ actual class StringResource(
         val string = resourceBundle.getString(key)
         return String.format(locale = locale, string, *args)
     }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is StringResource) return false
+        if (key != other.key) return false
```
Comment (Alex009) [inline resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt:28]: bundleName should be compared too

Comment (Alex009) [inline resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt:28]: bundleName should be compared too

File: resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt [hunk 2]
```diff
@@ -22,4 +21,13 @@ actual class StringResource(
...
+        return true
+    }
+
+    override fun hashCode() = key.hashCode()
```
Comment (Alex009) [inline resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt:32]: and bundleName too

Comment (Alex009) [inline resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt:32]: and bundleName too

**Review by Alex009 — CHANGES_REQUESTED** [review]

File: resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt (hunk 1 above)
Comment (y9san9) [reply resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt]: now it's generated by data class

File: resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt (hunk 2 above)
Comment (y9san9) [reply resources/src/jvmMain/kotlin/dev/icerock/moko/resources/StringResource.kt]: now it's generated by data class

**Review by Alex009 — APPROVED** [review]

**Pull request merged**